### PR TITLE
topology: revert apl-nocodec to static pipelines for now

### LINKED ISSUE #4199: [BUG]DSP panic and dai_config timeout on APL_UP2_NOCODEC
**Describe the bug**
DSP panic on APL_UP2_NOCODEC ,causing multiple-pause-resume-25  to fail  since May 13
eg. internal daily 4313,4300
in case 'multiple-pause-resume-25'

**To Reproduce**
Reproduced in CI daily test since May 13)
TPLG=sof-apl-nocodec.tplg ~/sof-test/test-case/multiple-pause-resume.sh -r 25

**Environment**
Kernel Branch: topic/sof-dev
Kernel Commit: caca1987
SOF Branch: main
SOF Commit: 9235eb83608f

**Log**
[console]
```
[  171.902145] kernel: sof-audio-pci-intel-apl 0000:00:0e.0: error : DSP panic!
[  171.902162] kernel: sof-audio-pci-intel-apl 0000:00:0e.0: status: fw entered - code 00000005
[  171.902208] kernel: sof-audio-pci-intel-apl 0000:00:0e.0: error: runtime exception
[  171.902212] kernel: sof-audio-pci-intel-apl 0000:00:0e.0: error: trace point 00004000
[  171.902216] kernel: sof-audio-pci-intel-apl 0000:00:0e.0: error: panic at :0
[  171.902220] kernel: sof-audio-pci-intel-apl 0000:00:0e.0: error: DSP Firmware Oops
[  171.902224] kernel: sof-audio-pci-intel-apl 0000:00:0e.0: error: Exception Cause: LoadStorePIFDataErrorCause, Synchronous PIF data error during LoadStore access
```
[dmesg log]
```
[  171.902145] sof-audio-pci-intel-apl 0000:00:0e.0: error : DSP panic!
[  171.902156] sof-audio-pci-intel-apl 0000:00:0e.0: panic: dsp_oops_offset 788480 offset 788480
[  171.902162] sof-audio-pci-intel-apl 0000:00:0e.0: status: fw entered - code 00000005
[  171.902208] sof-audio-pci-intel-apl 0000:00:0e.0: error: runtime exception
[  171.902212] sof-audio-pci-intel-apl 0000:00:0e.0: error: trace point 00004000
[  171.902216] sof-audio-pci-intel-apl 0000:00:0e.0: error: panic at :0
[  171.902220] sof-audio-pci-intel-apl 0000:00:0e.0: error: DSP Firmware Oops
[  171.902224] sof-audio-pci-intel-apl 0000:00:0e.0: error: Exception Cause: LoadStorePIFDataErrorCause, Synchronous PIF data error during LoadStore access
[  171.902228] sof-audio-pci-intel-apl 0000:00:0e.0: EXCCAUSE 0x0000000d EXCVADDR 0x0000005c PS       0x00060925 SAR     0x00000000
[  171.902233] sof-audio-pci-intel-apl 0000:00:0e.0: EPC1     0xbe0096d3 EPC2     0xbe0091aa EPC3     0x00000000 EPC4    0x00000000
[  171.902238] sof-audio-pci-intel-apl 0000:00:0e.0: EPC5     0x00000000 EPC6     0x00000000 EPC7     0x00000000 DEPC    0x00000000
[  171.902242] sof-audio-pci-intel-apl 0000:00:0e.0: EPS2     0x00060d20 EPS3     0x00000000 EPS4     0x00000000 EPS5    0x00000000
[  171.902247] sof-audio-pci-intel-apl 0000:00:0e.0: EPS6     0x00000000 EPS7     0x00000000 INTENABL 0x00000000 INTERRU 0x00000262
[  171.902251] sof-audio-pci-intel-apl 0000:00:0e.0: stack dump from 0xbe047540
[  171.902257] sof-audio-pci-intel-apl 0000:00:0e.0: 0xbe047540: be047580 be068478 be068478 be04d0e4
[  171.902262] sof-audio-pci-intel-apl 0000:00:0e.0: 0xbe047544: 7e0203bb be0475c0 00000000 00000001
[  171.902267] sof-audio-pci-intel-apl 0000:00:0e.0: 0xbe047548: 3fe76b00 047a3d6e 000c0800 00000000
[  171.902292] sof-audio-pci-intel-apl 0000:00:0e.0: 0xbe04754c: 0dead000 00000000 84066828 ffff991b
[  171.902297] sof-audio-pci-intel-apl 0000:00:0e.0: 0xbe047550: c074903f ffffffff 85082800 ffff991b
[  171.902301] sof-audio-pci-intel-apl 0000:00:0e.0: 0xbe047554: 85079f00 ffff991b 8148c408 ffffffff
[  171.902306] sof-audio-pci-intel-apl 0000:00:0e.0: 0xbe047558: 83b59dc0 ffff991b 814cfa80 ffffffff
[  171.902311] sof-audio-pci-intel-apl 0000:00:0e.0: 0xbe04755c: 83b59dc0 ffff991b 00000000 00000000
[  171.902315] sof-audio-pci-intel-apl 0000:00:0e.0: error: waking up any trace sleepers
[  171.902447] sof-audio-pci-intel-apl 0000:00:0e.0: error: trace IO error

```

### PR DESCRIPTION
**Pull request by marc-hb**
This partial, apl-nocodec only revert of commit 426d7816e265 ("topology:
nocodec: switch to dynamic pipelines") solves the `aplay:
set_params:1432: Unable to install hw params:` failure described below.

According to Xiaoyun, it also fixes #4199. This regression has been
opened for more than 2 weeks now.

Note `-DDYNAMIC=0` produces the same output as `-DDYNAMIC=1`. Bug or
feature? EDIT: apparently an `m4` "feature". In that case I recommend `-DDYNAMIC=some_thing`

Reproduction steps:

- Unload the drivers with `sof-test/tools/kmod/sof_remove.sh`

- Insert the drivers and REQUIRED: turn off Power Management [as described in sof-docs](https://thesofproject.github.io/latest/developer_guides/debugability/logger/index.html)

```
sof-test/tools/kmod/sof_insert.sh && echo on |   sudo tee /sys/devices/pci0000:00/0000:00:0e.0/power/control > /dev/null
```

EDIT+ WARNING: all the above must be on the same line NOT to give the DSP time to be suspended which happens after a couple seconds. Tip: `journalctl -f -p 7 -g 'DSP'`

In other words this bug reproduces only on the very first DSP boot.

- sof-test/test-case/check-playback.sh -d 1 -l 1
```
  aplay   -Dhw:0,1 -r 48000 -c 2 -f S16_LE -d 1 /dev/zero -v -q
  aplay: set_params:1432: Unable to install hw params:
```
Trace shows `ERROR ipc: comp X pipeline not found`

Notes:

- aplay seems to fail only once per pipeline. After failing the first
  time, each pipeline seems to be working fine.

- Turning off power management seems required.

Signed-off-by: Marc Herbert <marc.herbert@intel.com>

### CODE REVIEW

**Comment by ranj063** [comment]
@marc-hb what are the odds somebody will fix these issues if you revert dynamic pipelines for this tplg? If we really want to show a green, can you please escalate the priority for the bug instead?

**Review by kv2019i — CHANGES_REQUESTED** [review]
Let's try to address this via priorities. The affected bug did not have any priority set and nobody was assigned to fix it. I now tagged it as P1 and assigned it to someone. If it still remains open, then we can consider reverting, but let's try. We had to back down from dynamic pipelines already for 1.8 and I think same will happen for 1.9 unless we fix the issues.

**Review by plbossart — APPROVED** [review]
I highlighted validation issues last week, nothing happened.
I have no issue with reverting a single commit that clearly needs more time.

**Comment by marc-hb** [comment]
> I have no issue with reverting a single commit that clearly needs more time.

It's even less than a single commit BTW: it's only one line / one platform from it.

**Comment by ranj063** [comment]
Please dont merge this. I am close to finding a fix for #4199

**Comment by marc-hb** [comment]
After an intense pair-debugging session with @ranj063 (thanks!) where she could desperately not reproduce I found this critical reproduction requirement, which I just added to the description (not to the commit message):

- Insert the drivers and REQUIRED: turn off Power Management [as described in sof-docs](https://thesofproject.github.io/latest/developer_guides/debugability/logger/index.html)

```
sof-test/tools/kmod/sof_insert.sh && echo on |  sudo tee /sys/devices/pci0000:00/0000:00:0e.0/power/control 
```

EDIT+ WARNING: all the above must be on the same line NOT to give the DSP time to be suspended which happens after a couple seconds. Tip: `journalctl -f -p 7 -g 'DSP'`

In other words this bug reproduces *only on the very first DSP boot.*

**Comment by marc-hb** [comment]
> Please dont merge this. I am close to finding a fix for #4199

I tested #4282 and it does fix #4199 in my local testing but it does _not_ fix this here.

**Comment by ranj063** [comment]
@marc-hb I think it is an artificial bug caused due to timing issues between sof_insert and disabling PM while during sof_insert.

sof-test/tools/kmod/sof_insert.sh && echo on |  sudo tee /sys/devices/pci0000:00/0000:00:0e.0/power/control 

@plbossart do you think we should worry about this?

**Comment by marc-hb** [comment]
This PR is turning into a bug, I will do a bit more testing and file a new bug later. It will probably be called something like "dynamic topologies now require one initial D3 suspend cycle"

**Comment by ranj063** [comment]
> This PR is turning into a bug, I will do a bit more testing and file a new bug later. It will probably be called something like "dynamic topologies now require one initial D3 suspend cycle"

here's the fix for the problem https://github.com/thesofproject/linux/pull/2964

**Comment by lgirdwood** [comment]
@marc-hb any update here ? I guess we have some recent fixes merged now so is this issue fixed for you ?

**Comment by marc-hb** [comment]
> here's the fix for the problem thesofproject/linux#2964

I cherry-picked this and fix confirmed, thanks!

**Comment by marc-hb** [comment]
> @marc-hb any update here ? I guess we have some recent fixes merged now so is this issue fixed for you ?

I downgraded this PR to a github draft when @ranj063 announced a fix. Drafts appear grey instead of green, that's a bit too subtle, next time I'll add a `[DRAFT]` prefix. There's also the very useful:

https://github.com/thesofproject/sof/pulls?q=is%3Apr+is%3Aopen+draft%3Afalse

**Comment by marc-hb** [comment]
Was re-submitted and merged in commit 5b072d97d5 / PR #4323
